### PR TITLE
[ODS-4534] Deployment for .NET Core Sandbox Admin

### PR DESCRIPTION
**Pull request by tnunnerydl**

### CODE REVIEW

File: Application/EdFi.Ods.SandboxAdmin/EdFi.Ods.SandboxAdmin.nuspec [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+    <description>EdFi.Ods.SandboxAdmin</description>
     <releaseNotes></releaseNotes>
     <license type="file">LICENSE.txt</license>
   </metadata>
   <files>
-    <file src="../../LICENSE.txt" />
-    <file src="../../NOTICES.md" />
+    <file src="../../../LICENSE.txt" />
+    <file src="../../../NOTICES.md" />
     <file src="**" />
```
Comment (jamessmckay) [inline Application/EdFi.Ods.SandboxAdmin/EdFi.Ods.SandboxAdmin.nuspec:17]: we should consider excluding appsettings.development.json and appsettings.user.json

File: Application/SolutionScripts/InitializeDevelopmentEnvironment.psm1 [hunk 2]
```diff
@@ -225,7 +225,7 @@ function Add-SandboxCredentials {
         [string] $populatedSecret = (Get-RandomString),
         [string] $minimalKey = (Get-RandomString),
         [string] $minimalSecret = (Get-RandomString),
-        [string] $adminCredentialConfigPath = (Join-Path (Get-RepositoryRoot $implementationRepo) 'Application\EdFi.Ods.SandboxAdmin.Web\AdminCredential.config'),
+        [string] $adminCredentialConfigPath = (Join-Path (Get-RepositoryRoot $implementationRepo) 'Application\EdFi.Ods.SandboxAdmin\AdminCredential.config'),
```
Comment (jamessmckay) [inline Application/SolutionScripts/InitializeDevelopmentEnvironment.psm1:228]: should this be the appsettings.user.json instead?

File: Application/Ed-Fi-Ods.sln [hunk 3]
```diff
@@ -1,405 +1,396 @@
...
-		{658A15C4-A69B-4825-BE4B-9B3F37FA7FA2} = {F25ED154-FB97-4F39-933B-808D1B3B6D3E}
-		{37CB77F7-1C51-480C-B536-CDBF5DE26E10} = {14352B5E-2434-4117-A063-2BBD4D87464D}
-		{11894549-6088-46EA-BBD9-52F254E70FDA} = {26FBA24F-2F1E-47EA-BDE2-EF57AE81D65D}
-		{D2E6E31D-FA21-4CF6-8C59-CF69BE4FDD50} = {26FBA24F-2F1E-47EA-BDE2-EF57AE81D65D}
-	EndGlobalSection
-	GlobalSection(ExtensibilityGlobals) = postSolution
-		SolutionGuid = {91F27EA2-CFF0-4054-A019-B8AA23AF9319}
-	EndGlobalSection
-EndGlobal
+Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (jamessmckay) [inline Application/Ed-Fi-Ods.sln:1]: curious why is this whole file changed?

Comment (jamessmckay) [reply Application/Ed-Fi-Ods.sln:1]: also are we ready to remove the old sandbox admin application?

**Review by jamessmckay — CHANGES_REQUESTED** [review]
Are we ready to remove the old application for sandbox?

File: Application/SolutionScripts/InitializeDevelopmentEnvironment.psm1 (hunk 2 above)
Comment (tbmeador-dlp) [reply Application/SolutionScripts/InitializeDevelopmentEnvironment.psm1:228]: This function is deprecated and can be removed from running as part of initdev. Let's make sure these settings are now correctly coming from the settings-management.psm1

Comment (tnunnerydl) [reply Application/SolutionScripts/InitializeDevelopmentEnvironment.psm1:226]: I removed the call to this function in initdev.

File: logistics/scripts/modules/settings/settings-management.psm1 [hunk 4]
```diff
@@ -322,11 +324,81 @@ function New-DevelopmentAppSettings([hashtable] $Settings = @{ }) {
...
-        New-JsonFile $newUserSettingsPath $credentialSettings
+        New-JsonFile $newUserSettingsPath $credentialSettings -Overwrite
         $newSettingsFiles += $newUserSettingsPath
     }
 
     return $newSettingsFiles
 }
+
+function Format-Json {
+    <#
```
Comment (tnunnerydl) [inline logistics/scripts/modules/settings/settings-management.psm1:337]: Found this function in stackoverflow to format JSON files.  If you have another option, please let me know.

Comment (jamessmckay) [reply logistics/scripts/modules/settings/settings-management.psm1:337]: Very interesting

File: Application/EdFi.Ods.SandboxAdmin/appsettings.json [hunk 5]
```diff
@@ -19,6 +19,7 @@
   "DefaultOperationalContextUri": "uri://ed-fi-api-host.org",
   "OAuthUrl": "http://localhost:54746/oauth/",
```
Comment (jamessmckay) [inline Application/EdFi.Ods.SandboxAdmin/appsettings.json:20]: should this be empty and added in the developers settings file?

**Review by jamessmckay — CHANGES_REQUESTED** [review]

File: Application/EdFi.Ods.SandboxAdmin/EdFi.Ods.SandboxAdmin.nuspec (hunk 1 above)
Comment (tnunnerydl) [reply Application/EdFi.Ods.SandboxAdmin/EdFi.Ods.SandboxAdmin.nuspec:17]: Files excluded

File: Application/Ed-Fi-Ods.sln (hunk 3 above)
Comment (tnunnerydl) [reply Application/Ed-Fi-Ods.sln:1]: As far as I know, we are ready to remove sandboxAdmin.Web.

File: Application/EdFi.Ods.SandboxAdmin/appsettings.json (hunk 5 above)
Comment (tnunnerydl) [reply Application/EdFi.Ods.SandboxAdmin/appsettings.json]: Done.

File: Application/EdFi.Ods.SandboxAdmin/appsettings.development.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
+﻿{
```
Comment (jamessmckay) [inline Application/EdFi.Ods.SandboxAdmin/appsettings.development.json:1]: This file should not be checked in

File: Application/EdFi.Ods.SandboxAdmin/appsettings.json [hunk 7]
```diff
@@ -10,18 +16,22 @@
...
+  },
+  "User": {},
+    "MailSettings": {
+      "Smtp": {
+        "UserName": "Bingo",
+        "Password": "Tingo",
+        "DeliveryMethod": "SpecifiedPickupDirectory",
+        "From": "noreply@ed-fi.org",
+        "SpecifiedPickupDirectory": {
+          "PickupDirectoryLocation": "C:\\Temp\\AdminConsole\\Artifacts\\Emails"
```
Comment (jamessmckay) [inline Application/EdFi.Ods.SandboxAdmin/appsettings.json:33]: should the mail settings be set to empty and updated in the appsettings.user.json?

File: Application/SolutionScripts/InitializeDevelopmentEnvironment.psm1 [hunk 8]
```diff
@@ -225,7 +223,7 @@ function Add-SandboxCredentials {
...
```
Comment (jamessmckay) [inline Application/SolutionScripts/InitializeDevelopmentEnvironment.psm1:226]: should we be using the appsettings.user.json that is generated here?

File: logistics/scripts/modules/settings/settings-management.psm1 [hunk 9]
```diff
@@ -199,7 +199,7 @@ function New-JsonFile {
 
     if (-not $Overwrite -and (Test-Path $FilePath)) { return }
 
-    $Hashtable | ConvertTo-Json -Depth 10 | Out-File -FilePath $FilePath -NoNewline -Encoding UTF8
+    $Hashtable | ConvertTo-Json -Depth 10 | Format-Json |  Out-File -FilePath $FilePath -NoNewline -Encoding UTF8
```
Comment (jamessmckay) [inline logistics/scripts/modules/settings/settings-management.psm1:202]:
```suggestion
    $Hashtable | ConvertTo-Json -Depth 10 | Format-Json | Out-File -FilePath $FilePath -NoNewline -Encoding UTF8
```

**Review by jamessmckay — CHANGES_REQUESTED** [review]

File: Application/SolutionScripts/InitializeDevelopmentEnvironment.psm1 (hunk 8 above)
Comment (tnunnerydl) [reply Application/SolutionScripts/InitializeDevelopmentEnvironment.psm1:226]: This function is no longer used and will be removed on the next release.

**Review by jamessmckay — APPROVED** [review]

**Review by jamessmckay — APPROVED** [review]

**Pull request merged**